spec/models/course_spec.rb

```diff
@@ -2135,6 +2135,119 @@
+    let(:organisation) { create(:organisation, providers: [provider]) }
+    let(:site_status) { create(:site_status, :findable) }
+    let(:user_one) { create(:user, organisations: [organisation]) }
+    let(:user_two) { create(:user, organisations: [organisation]) }
+
+    let(:course) do
+      create(
+        :course,
+        provider: provider,
+        accrediting_provider_code: accrediting_provider.provider_code,
+        age_range_in_years: "11_to_15",
+        qualification: "pgce_with_qts",
+        study_mode: "full_time",
+        site_statuses: [site_status],
+        maths: :equivalence_test,
+        english: :equivalence_test,
+      )
+    end
+
+    let(:mail_spy) { spy }
+    let(:mailer_spy) { spy(create_course_email: mail_spy) }
+
+    before do
+      stub_const("CreateCourseEmailMailer", mailer_spy)
+    end
+
+    context "A self-accredited course" do
+      let(:course) { create(:course, :self_accredited, provider: accrediting_provider) }
+
+      before do
+        UserNotification.new(user: user_one, provider_code: accrediting_provider.provider_code, create_course: true).save!
+        UserNotification.new(user: user_two, provider_code: accrediting_provider.provider_code, create_course: false).save!
+      end
+
+      it "Does not send a notification" do
+        course
+
+        expect(mailer_spy).not_to have_received(:create_course_email)
+      end
+    end
+
+    context "A non self-accredited course" do
+      context "With no users with notifications enabled" do
+        it "does nothing" do
+          course
+
+          expect(mailer_spy).not_to have_received(:create_course_email)
+        end
+      end
+
+      context "With a user with notifications enabled" do
+        before do
+          UserNotification.new(user: user_one, provider_code: accrediting_provider.provider_code, create_course: true).save!
+          UserNotification.new(user: user_two, provider_code: accrediting_provider.provider_code, create_course: false).save!
+          course
+        end
+
+        it "Sends the notification to the correct user" do
+          expect(mailer_spy).to have_received(:create_course_email) do |course, user|
+            expect(course).to eq(course)
+            expect(user).to eq(user_one)
+          end
+        end
+
+        it "Delivers the email" do
+          expect(mail_spy).to have_received(:deliver_now)
+        end
+
+        context "When the course does not appear on find" do
+          let(:site_status) { create(:site_status, :unpublished) }
+
+          it "Does not send a notification" do
+            course
+
+            expect(mailer_spy).not_to have_received(:create_course_email)
+          end
+        end
+      end
+      context "With multiple users with notifications enabled" do
+        before do
+          UserNotification.new(user: user_one, provider_code: accrediting_provider.provider_code, create_course: true).save!
```

Comment:
We should create a factory for these. That said, if we add the scope we can stub it here rather than have to involve the database (as long as this is covered in a higher level integration test somewhere)